src/main/java/com/gluonhq/substrate/target/LinuxLinkerFlags.java

```diff
@@ -0,0 +1,190 @@
+ */
+package com.gluonhq.substrate.target;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.gluonhq.substrate.target.LinuxFlavor.Flavor;
+import com.gluonhq.substrate.util.Logger;
+import com.gluonhq.substrate.util.ProcessRunner;
+
+import static com.gluonhq.substrate.target.LinuxLinkerFlags.PkgInfo.activeOf;
+import static com.gluonhq.substrate.target.LinuxLinkerFlags.PkgInfo.debian;
+import static com.gluonhq.substrate.target.LinuxLinkerFlags.PkgInfo.fedora;
+import static com.gluonhq.substrate.target.LinuxLinkerFlags.PkgInfo.hardwired;
+
+/**
+ * Defines linker flags for a given linux flavor (debian/fedora).
+ */
+public class LinuxLinkerFlags {
+    private static final Flavor flavor = new LinuxFlavor().getFlavor();
+
+    /**
+     * Defines per-flavor pkg-config package names and the associated OS package
+     * name that provides it.
+     */
+    private static final List<PkgInfo> LINK_DEPENDENCIES = List.of(
+        hardwired("-Wl,--no-whole-archive"),
+
+        activeOf(debian("gl", "libgl-dev"),
+                 fedora("gl", "mesa-libGL-devel")),
+        activeOf(debian("x11", "libx11-dev"),
+                 fedora("x11", "libX11-devel")),
+        activeOf(debian("gtk+-x11-3.0", "libgtk-3-dev"),
+                 fedora("gtk+-3.0", "gtk3-devel")),
+        activeOf(debian("freetype2", "libfreetype6-dev"),
+                 fedora("freetype2", "freetype-devel")),
+        activeOf(debian("pangoft2", "libpango1.0-dev"),
+                 fedora("pangoft2", "pango-devel")),
+
+        hardwired("-lgstreamer-lite"),
+
+        activeOf(debian("gthread-2.0", "libglib2.0-dev"),
+                 fedora("gthread-2.0", "glib2-devel")),
+
+        hardwired("-lstdc++"),
+        
+        activeOf(debian("zlib", "zlib1g-dev"),
+                 fedora("zlib", "zlib-devel")),
+
+        activeOf(debian("xtst", "libxtst-dev"),
+                 fedora("xtst", "libXtst-devel")),
+
+        // On fedora these require https://rpmfusion.org/
+        activeOf(debian("libavcodec", "libavcodec-dev"),
+                 fedora("libavcodec", "ffmpeg-devel")),
+        activeOf(debian("libavformat", "libavformat-dev"),
+                 fedora("libavformat", "ffmpeg-devel")),
+        activeOf(debian("libavutil", "libavutil-dev"),
+                 fedora("libavutil", "ffmpeg-devel")),
+
+        activeOf(debian("alsa", "libasound2-dev"),
+                 fedora("alsa", "alsa-lib-devel")),
+
+        hardwired("-lm"),
+
+        activeOf(debian("gmodule-no-export-2.0", "libglib2.0-dev"),
+                 fedora("gmodule-no-export-2.0", "glib2-devel"))
+    );
+
+    public List<String> getLinkerFlags() {
+        List<String> missingPackages = new ArrayList<>();
+
+        List<String> pkgFlags = LINK_DEPENDENCIES.stream()
+                .flatMap(pkg -> lookupPackageFlags(pkg, missingPackages).stream())
```

Comment:
At some point we should cache this check, once all the packages are found, it doesn't make sense to check on every run, does it? It could be done in a follow-up PR, of course.